File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int key = 0;
    int val = -1;
    public int kthSmallest(TreeNode root, int k) {
        key = k;
        helper(root);
        return val;
    }

    private void helper(TreeNode root) {
        if (root == null) return;
        helper(root.left);
        if (key-- == 1) {
            val = root.val;
            return;
        }
        helper(root.right);
    }


}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        queue = deque()
        result = []

        if root:
            queue.append(root)

        while len(queue) > 0:
            tempList = []
            for _ in range(len(queue)):
                node = queue.popleft()
                tempList.append(node.val)
                if node.left: 
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            result.append(tempList[-1])

        return result
